lib/pipeline.js

```diff
@@ -301,6 +303,47 @@ class PipelineModel extends BaseModel {
                         throw new Error(message);
                     });
                 })
+                .then(screwdriverConf => {
+                    if (pipelineFactory.isExtendingEnabled()) {
```

Comment:
this is part of `pipeline.sync` which happens for all event creation or webhook delivery. and will be a waste of processing time. Webhooks already have strict sla of < 10s. So I would prefer to have this generation an  offline process, which is not part of run time.

Reply:
How much time/ Error handling cases can be tuned further using HTTP API timeouts, and defaults for the API configs. But again limiting everywhere on the user is what the first reason for other choices. Think from the user perspective. If there is a strong reason that we have to finish the total process in a certain time we can cap the max timeout for this API in the code. But there should be a reason why we caping it. And making the user aware about the impact of long HTTP timeout is a better way to go ahead is what my kind opinion.

Reply:
we don't have control of this, this is GHE webhook time out. So we have to be mindful of not breaking webhook processing. We cab't penalize users for this, so our deisgn has to take into account on this limit. 
We do have plans to make entire webhook processing an offline process. https://github.com/screwdriver-cd/screwdriver/issues/1769

Reply:
My point was more on the lines of, instead of generating this dynamically during each build run, this screwdrive.yaml config can be generated when ever there is a change in config per repo or there is a change in upstream process which defines overall config.